src/ngraph/op/topk.cpp

```diff
@@ -32,8 +32,9 @@ op::TopK::TopK(const Output<Node>& arg,
                size_t k,
                bool compute_max,
                SortType sort)
-    : Op({arg, op::Constant::create(element::i64, Shape{1}, {k})->output(0)})
-    , m_top_k_axis(top_k_axis)
+    : Op({arg,
+          op::Constant::create(element::i64, Shape{1}, {k})->output(0),
+          op::Constant::create(element::i64, Shape{1}, {top_k_axis})->output(0)})
     , m_index_element_type(index_element_type)
```

Comment:
These constants (and in the other constructor) should be set up in `validate_and_infer_types`. 
All created constants need to be be added to the node's provenance group.

Reply:
I updated provenance. Don't quite get "set up the constant in `validate_and infer_types`". Can you check?

Reply:
Don't create the constant in the constructor. Instead, validate_and_infer_types should create them if the inputs are missing. So here you would do `Op({arg})` and in validate_and_infer_types if the number of args is 1 then create the constants. You'll need to stick `k` in a member variable to get it from the constructor to validate.